### DISCUSSION TITLE
Add method to update multiple pivot records at once

### DISCUSSION THREAD

**Discussion by Synchro** [laravel/framework#47103] [Ideas]
At present, if you want to update multiple (typically all of them) non-id fields in pivot records, you need to do this:

```php
$user = User::find(1);
 
foreach ($user->roles as $role) {
    $user->roles()->updateExistingPivot(
        $role->id,
        [
            'active' => false,
        ]
    );
}
```
This is inefficient as it issues a query for every record (a classic N+1). It would be nice to have a method that updates all pivot records in one go, something like this:

```php
$user->roles()->updateExistingPivots(['active' => false,]);
```

Until then we can either live with the overhead, or resort to a manually constructed query.

**Comment by RTippin** [comment]
@Synchro  I believe you could use `newPivotQuery` in the chain:

```php
$user->roles()->newPivotQuery()->update(['active' => false]);
```
